src/trip_kinematics/URDFParser.py

```diff
@@ -0,0 +1,301 @@
+    to (0, 0, 0). <axis> defaults to (0, 0, 1).
+
+    Args:
+        filename (str): Path to URDF file.
+
+    Raises:
+        ValueError: Could not parse URDF file.
+
+    Returns:
+        List[Transformation]: List of transformations that describe the robot.
+    """
+    tree = ET.parse(filename)
+    root = tree.getroot()
+    joints = root.findall('joint')
+    joint_name_to_transformations = {}
+
+    joint_tree_dict = _build_joint_tree_dict(joints)
+
+    root_joints = [name for name, value in joint_tree_dict.items() if not value['parent']]
+
+    for i, joint in enumerate(joints):
+        name = joint.get('name')
+        if not name:
+            raise ValueError('Missing name field in URDF file')
+        joint_tree_dict[name]['joint_index'] = i
+
+    for joint in joints:
+        joint_name_to_transformations[joint.get('name')] = _get_transformations_for_joint(joint)
+
+    transformations = []
+    for joint in root_joints:
+        transformations.extend(_create_transformations_from_tree(joint,
+                                                                 joint_tree_dict,
+                                                                 joint_name_to_transformations,
+                                                                 None)
+                               )
+
+    return transformations
+
+
+def align_vectors(target: np.ndarray, to_align: np.ndarray) -> np.ndarray:
+    """Generates a rotation matrix that makes b parallel to a.
+
+    Args:
+        a (np.ndarray): 3D Vector.
+        b (np.ndarray): 3D Vector.
+
+    Returns:
+        np.ndarray: 3x3 rotation matrix.
+    """
+    target = target / np.linalg.norm(target)
+    to_align = to_align / np.linalg.norm(to_align)
+
+    # If the vectors are parallel but in opposite directions, return a 180 degree rotation
+    if np.array_equal(target, -to_align):
+        return -np.identity(3)
+
+    align_axis = np.cross(target, to_align)
+    cos_angle = np.dot(target, to_align)
+    k = 1 / (1 + cos_angle)
+
+    result = np.array([[(align_axis[0] * align_axis[0] * k) + cos_angle,
+                        (align_axis[1] * align_axis[0] * k) - align_axis[2],
+                        (align_axis[2] * align_axis[0] * k) + align_axis[1]],
+                       [(align_axis[0] * align_axis[1] * k) + align_axis[2],
+                        (align_axis[1] * align_axis[1] * k) + cos_angle,
+                        (align_axis[2] * align_axis[1] * k) - align_axis[0]],
+                       [(align_axis[0] * align_axis[2] * k) - align_axis[1],
+                        (align_axis[1] * align_axis[2] * k) + align_axis[0],
+                        (align_axis[2] * align_axis[2] * k) + cos_angle]])
+    return result
+
+
+def _build_joint_tree_dict(joints: List[ET.Element]) -> Dict[str, Dict]:
+    """Creates a dictionary representing parent-child relationships between joints. Used by
+    from_urdf() to build a tree of joints.

```

Comment:
Sphinx allows using special tags that make the reference to 'from_urdf()' into a dynamic reference.
More information can be found [here](https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#cross-referencing-syntax)
Please use these tags